platform/android/java/editor/src/main/java/org/godotengine/editor/GradleBuildProvider.kt

```diff
@@ -0,0 +1,86 @@
+/**************************************************************************/
+/*  GradleBuildProvider.kt                                                */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+package org.godotengine.editor
+
+import org.godotengine.godot.Godot
+import org.godotengine.godot.BuildProvider
+import org.godotengine.godot.variant.Callable
+
+internal class GradleBuildProvider(
+	private val gradleBuildEnvironmentClient: GradleBuildEnvironmentClient,
```

Comment:
Let's merge those two classes. Or if it's cleaner to keep them separate, let's create a `buildprovider` package and move both classes in there.

If we keep them separate, `GradleBuildProvider` should be the driver (i.e. create and setup `GradleBuildEnvironmentClient` by itself).

Reply:
It's fine to keep them separate! In that case let's move them to their own `buildprovider` package. 
I want to encapsulate `GradleBuildEnvironmentClient` away from `BaseGodotEditor`. 

`BaseGodotEditor` should only interact with `GradleBuildProvider` and the latter does whatever is necessary behind the scene (aka init and use `GradleBuildEnvironmentClient`) to set up the build environment.

Reply:
I've moved them into their own package!